inc/datahandlers/post.php

```diff
@@ -1255,6 +1252,12 @@ function insert_post()
 			$thread_update = array('replies' => '+1');
 			update_forum_counters($post['fid'], array("deletedposts" => "+1"));
 		}
+		
+		// Update forum count
+		if($visible == 1) {
+			update_last_post($post['tid']);
+			update_forum_lastpost($thread['fid']);
```

Comment:
It's not necessary to move update_forum_lastpost. It doesn't need to be updated when the thread isn't visible.